### ISSUE TITLE
[Contribution] Android Method wish Import for Genshin Impact 3.2 (No pc,No root and no Wifi Debugging)

### ISSUE THREAD

**Issue by MizukumaMC** [MadeBaruna/paimon-moe#333]
Thanks me later paimon.moe devs

https://youtu.be/dCBPJmVL4HA

**Comment by Deses** [comment]
Cool, yet very convoluted method. I don't think the average user will be able to follow it, and I expect most phones to have different developer options. for instance, my ROM doesn't have that toggle to generate bug reports (Oneplus 7Pro running a custom AOSP ROM).

Also, there are a lot of improvements that could be made in the process show in your video:
1.- You don't need to email the zip to yourself. Just open "My Files" app and locate the zip.
2.- You don't need a dedicated unzip app, any modern "My Files" app should be able to open compressed files and if not, download a better one. Try `CX File Explorer` or `Amaze File Manager`.
3.- I know most default text apps can't open huge log files, but you don't need to install tmux to open the file and locate the URL string, there are more user friendly ways to do that, for instance, you could use `QuickEdit` to open the file and from there, copy the desired string.

**Comment by MizukumaMC** [comment]
Quick edit give me an annonying error that file is too big! Cuz termux `less` command is a unix file reader so it can read upto 200GB txt files

**Comment by MizukumaMC** [comment]
2. Z archiever is just an example and I'm not comfortable to use the file manager that came on my phone!

**Comment by MizukumaMC** [comment]
If you use custom ROM,try to use stock ROM

**Comment by MizukumaMC** [comment]
Btw thank you for feedback❤

**Comment by Deses** [comment]
> Quick edit give me an annonying error that file is too big! Cuz termux `less` command is a unix file reader so it can read upto 200GB txt files

Just how big is that log file?



> If you use custom ROM,try to use stock ROM

Yeah. Good one, but github is no place to tell jokes.

> 4 replies

You know you can edit posts to add further comments, right?

**Comment by fghzxm** [comment]
So if my understanding is correct, you basically inspect the system logs and search for an entry tagged `MiHoYoWebview.load(L:90)`.  Using `adb` on my computer through USB debugging, I can do that with one command:
```bash
adb logcat -d -b main -v tag | grep --color=never -m1 -xe 'D/MiHoYoWebview\.load(?:\([^)]*\))?: load:.*'
```
This will grep out the entire line of log, not just the URL.  This method as it is does require you to do it on a computer; maybe using wireless debugging you can to do this on the phone itself, I haven't tried.

**Comment by jogerj** [comment]
Yes, we already have multiple variations of this solution in the discord server. I don't think there are plans to permanently have this method as the default. We frequently have inconsistent results with different brands of android phones, not to mention the difficulty for some people to follow

**Comment by fghzxm** [comment]
> We frequently have inconsistent results

Thanks for your feedback @jogerj.  Based on my testing, this method needs to be carried out fast enough after you enter the wishing history screen.  This is because the Android log buffer can be small, and Android can produce a _lot_ of logs every second which after some time will cause the log entry we're interested in to be purged.

If you remove the `-d` option to `adb`, it will keep reading new logs until <kbd>Ctrl</kbd>+<kbd>C</kbd> is pressed.  If you have time I suggest trying making this change then doing it _before_ you enter the wishing screen, and see if you can get better results.  Thanks.
